src/basic/string-table.h

```diff
@@ -29,7 +29,7 @@ ssize_t string_table_lookup(const char * const *table, size_t len, const char *k
 #define _DEFINE_STRING_TABLE_LOOKUP_FROM_STRING_WITH_BOOLEAN(name,type,yes,scope) \
         scope type name##_from_string(const char *s) {                  \
                 if (!s)                                                 \
-                        return -1;                                      \
+                        return -EINVAL;                                 \
```

Comment:
Currently, all `_XXX_INVALID` is mapped to `-EINVAL`, but I think it is better to use `_XXX_INVALID` instead of hard coded `-EINVAL`.

Reply:
We don't have access to `_XXX_INVALID` here. Those macros are called like `DEFINE_STRING_TABLE_LOOKUP(name_policy, NamePolicy)`, and we'd also need "NAME_POLICY".

Reply:
Yeah, indeed. `NAME_POLICY` is needed. Let's add the argument later if we have a case to set other errno than -EINVAL to `_XXX_INVALID`.